contrib/slurm/slurmexecutor.py

```diff
@@ -0,0 +1,297 @@
+        It also calls functions for output before and after the run.
+        """
+        self.output_handler.output_before_run(run)
+        benchmark = self.benchmark
+
+        args = run.cmdline()
+        logging.debug("Command line of run is %s", args)
+
+        try:
+            with open(run.log_file, "w") as f:
+                for i in range(6):
+                    f.write(os.linesep)
+
+            timelimit = self.benchmark.rlimits.cputime
+
+            run_result = run_slurm(
+                benchmark,
+                args,
+                run.log_file,
+                timelimit,
+                self.my_cpus,
+                benchmark.rlimits.memory,
+            )
+
+        except KeyboardInterrupt:
+            # If the run was interrupted, we ignore the result and cleanup.
+            stop()
+
+        if STOPPED_BY_INTERRUPT:
+            try:
+                if benchmark.config.debug:
+                    os.rename(run.log_file, run.log_file + ".killed")
+                else:
+                    os.remove(run.log_file)
+            except OSError:
+                pass
+            return 1
+
+        run.set_result(run_result)
+        self.output_handler.output_after_run(run)
+        return None
+
+
+def run_slurm(benchmark, args, log_file, timelimit, cpus, memory):
+    srun_timelimit_h = int(timelimit / 3600)
+    srun_timelimit_m = int((timelimit % 3600) / 60)
+    srun_timelimit_s = int(timelimit % 60)
+    srun_timelimit = f"{srun_timelimit_h}:{srun_timelimit_m}:{srun_timelimit_s}"
+
+    mem_per_cpu = int(memory / cpus / 1000000)
+
+    assert benchmark.config.scratchdir and os.path.exists(benchmark.config.scratchdir)
+
+    with tempfile.TemporaryDirectory(dir=benchmark.config.scratchdir) as tempdir:
+
+        os.makedirs(os.path.join(tempdir, "upper"))
+        os.makedirs(os.path.join(tempdir, "work"))
+
+        tool_command = " ".join(args)
+        singularity_command = (
+            f"singularity exec "
+            f"-B $PWD:/lower --no-home "
+            f"-B {tempdir}:/overlay "
+            f'--fusemount "container:fuse-overlayfs -o lowerdir=/lower -o upperdir=/overlay/upper -o workdir=/overlay/work $HOME" '
+            f"{benchmark.config.singularity} {tool_command}"
+            if benchmark.config.singularity
+            else tool_command
+        )
+        srun_command = (
+            f"srun "
+            f"-t {srun_timelimit} "
+            f"-c {cpus} "
+            f"-o {log_file} "
+            f"--mem-per-cpu {mem_per_cpu} "
+            f"--threads-per-core=1 "
+            f"--ntasks=1 "
+            f"{singularity_command}"
+        )
+        jobid_command = (
+            f"{srun_command} 2>&1 | grep -o 'job [0-9]* queued' | grep -o '[0-9]*'"
+        )
+        seff_command = f"seff $({jobid_command})"
+        logging.debug("Command to run: %s", seff_command)
+        result = subprocess.run(
+            ["bash", "-c", seff_command], shell=False, stdout=subprocess.PIPE
```

Comment:
What about `stderr`? Do you want `capture_output=True`?

Reply:
Right now, benchexec works for me with Python 3.6. `capture_output=True` would break this, so I'd rather avoid it. However, I'll make sure to capture stderr as well, that is indeed weird I haven't done.

Reply:
Komondor (which I realize is only relevant for me) has 3.6.8. Because right now benchexec works for me using this python version, I don't want to break this -- I do understand that other parts of Benchexec may already rely on later versions, and also that this could be modified any time, but I don't want to be the one to break it for myself.

Reply:
(I moved this comment from the PR's root thread to here, because I did not realize you answered here - github is very bad about loading new answers without a page reload today)

Reply:
I am somewhat surprised that they use 7 year old software on a cluster from last year - I hope for you that not the whole OS is that old, because I would expect that a kernel that old does not deliver optimal performance on an AMD Epyc 7763... But that is just a side note.

If you want to merge `stdout` and `stderr` into one stream, then `capture_output` is irrelevant anyway and there is no need to change anything.

But I still wanted to understand what your expectations and hopes are regarding us dropping support for older Python versions, because I don't want to be expected to keep BenchExec working on Python 3.6. If you are fine with it breaking, then we can move forward.

Our plans are also not set in stone, this is why we ask for feedback in #986. Feel free to comment there, in particular if you have a concrete request or suggestion, such as delaying the version bump for a specific time period or so if that would help you, or for looking at other distributions when determining which Python versions may still be relevant. We just don't want to keep support for old Python versions for a long time if nobody uses them anyway.

Reply:
Unfortunately, the software is quite old in some cases - the kernel is 4.18 (from 2018), which is far from optimal.

I'll copy my answer above to the issue you mentioned, but the worst that could happen for me is that I would need a local copy of a newer python version, which I don't think is that hard to achieve - however, I admit I have not yet tried to go down that route. So I don't want to be the one holding back benchexec with my (admittedly very niche) use-case.

Reply:
I've commented on the issue, I think we should discuss further caveats of python versioning there.